src/middlewares/authentication.js

```diff
@@ -0,0 +1,49 @@
+import { config } from 'dotenv';
+import jwt from 'jsonwebtoken';
+
+import database from '../models';
+import { ApplicationError } from '../helpers/errors';
+
+const { SECRET_KEY } = process.env;
+const { BlacklistedTokens } = database;
+
+config();
+
+export default {
+  /**
+   * Verify Token
+   *
+   * @param {Object} request - the request object
+   *
+   * @return {void} - undefined
+   */
+  verifyToken: async (request) => {
+    const authHeader = request.headers.authorization;
+    if (authHeader === '') throw new ApplicationError(400, 'No token provided. Please signup or login');
+    if (!authHeader) throw new ApplicationError(412, 'Header not set');
+    const token = authHeader.split(' ')[1];
+    const blacklistedToken = await BlacklistedTokens.findOne({
+      where: { token },
+    });
+
+    if (blacklistedToken) throw new ApplicationError(403, 'Please login or signup to access this resource');
+
+    jwt.verify(token, SECRET_KEY, (error, decodedToken) => {
+      if (error) throw new ApplicationError(401, `${error.message}`);
+      request.user = decodedToken;
+      return {};
```

Comment:
Why return an empty object?

Reply:
I just want to `return` the function at that point. But eslint is flagging the return statement saying it was unnecessary. I returned an empty object to satisfy eslint

Reply:
I just wanted to return the function at that point. I returned the empty object to avoid eslint from flagging it.